### PR TITLE
Add image feature extraction task spec and widget

### PR DESCRIPTION
**Pull request by merveenoyan**
cc @osanseviero as discussed

### CODE REVIEW

**Comment by merveenoyan** [comment]
unfortunately I linted and formatted yet there's no change I could stage in the files

**Comment by coyotte508** [comment]
The error is not in the lint but when building the types:

https://github.com/huggingface/huggingface.js/actions/runs/8613823630/job/23605959758?pr=614#step:6:104

> packages/tasks prepare: QuickTypeError: Key ImageClassificationParameters not in schema object at image-feature-extraction-input#ImageClassificationParameters

Probably this line: `			"$ref": "#/$defs/ImageClassificationParameters"` should be `			"$ref": "#/$defs/FeatureExtractionParameters"`

File: packages/tasks/src/tasks/image-feature-extraction/spec/input.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+			"type": "string"
+		},
+		"parameters": {
+			"description": "Additional inference parameters",
+			"$ref": "#/$defs/ImageClassificationParameters"
+		}
+	},
+	"$defs": {
+		"FeatureExtractionParameters": {
+			"title": "ImageFeatureExtractionParameters",
```
Comment (Wauplin) [inline packages/tasks/src/tasks/image-feature-extraction/spec/input.json:19]:
```suggestion
			"$ref": "#/$defs/ImageFeatureExtractionParameters"
		}
	},
	"$defs": {
		"ImageFeatureExtractionParameters": {
			"title": "ImageFeatureExtractionParameters",
```

**Comment by merveenoyan** [comment]
I have made a frankenstein widget from parts of image classification and feature extraction widget 🧟‍♂️ help is very appreciated!

**Comment by merveenoyan** [comment]
@mishig25 how can I get this up on my local to give you a screenshot? it should be text input and table output (like feature extraction but with image input)

**Comment by mishig25** [comment]
let me know if this readme insutrctions work https://github.com/huggingface/huggingface.js/tree/main/packages/widgets#demo

File: packages/tasks/src/tasks/image-feature-extraction/spec/input.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+	"$id": "/inference/schemas/image-feature-extraction/input.json",
+	"$schema": "http://json-schema.org/draft-06/schema#",
+	"description": "Inputs for Image Feature Extraction",
+	"title": "ImageFeatureExtractionInput",
+	"type": "object",
+	"properties": {
+		"inputs": {
+			"description": "The input image data",
+			"type": "string"
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-feature-extraction/spec/input.json:10]:
```suggestion
```

Comment (osanseviero) [reply packages/tasks/src/tasks/image-feature-extraction/spec/input.json:10]: Same type of input as other image tasks https://github.com/merveenoyan/huggingface.js/blob/add_ift_spec/packages/tasks/src/tasks/image-classification/spec/input.json#L8

**Comment by merveenoyan** [comment]
Closing this PR since widgets are moved back

**Pull request closed without merging**